lib/download.ex

```diff
@@ -116,8 +116,6 @@ defmodule Download do
 
   defp handle_async_response_chunk(%AsyncEnd{}, opts), do: finish_download({ :ok }, opts)
 
-  # Uncomment one line below if you are prefer to test not "Content-Length" header response, but a real file size
```

Comment:
The only problem that I found.

Please write some tests for this case - when `Content-Length` header and real body size aren't match.
Ok, we both know that right now we have `httpoison` behind `download` interface.

But what will be if we replace `httpoison` with, saying, `httpc`? Do you know how will it behave this case?

This case have to be tested too.

Reply:
Hey there, looking to get this package updated as I can definitely use a powerful but easy to use download library!

I'm not @little-bobby-tables, but it looks like the test [here](https://github.com/asiniy/download/pull/2/files#diff-a61b801e8dbe5c11dd374c223f983765R51) tests when the Content-Length header and real body size don't match- it downloads only the Content-Length, without anything else. Is another behavior desirable?

I'm not understanding the second point:

> Ok, we both know that right now we have httpoison behind download interface.

> But what will be if we replace httpoison with, saying, httpc? Do you know how will it behave this case?

> This case have to be tested too.

Nothing in his test code really relies on the library being HTTPoison, at least in this PR. What were you looking for here?